app/src/main/java/com/vestrel00/daggerbutterknifemvp/AppComponent.java

```diff
@@ -20,28 +20,25 @@
 
 import javax.inject.Singleton;
 
-import dagger.BindsInstance;
 import dagger.Component;
+import dagger.android.AndroidInjector;
+import dagger.android.support.AndroidSupportInjectionModule;
 
 /**
  * Injects application dependencies.
  */
 @Singleton
-@Component(modules = AppModule.class)
-interface AppComponent {
-    void inject(App app);
-
+@Component(modules = {
+        AndroidSupportInjectionModule.class,
+        AppModule.class
+})
+interface AppComponent extends AndroidInjector<App> {
```

Comment:
I like this solution better than [the one I found for providing the `Application` instance](https://github.com/vestrel00/android-dagger-butterknife-mvp/pull/44#issuecomment-320503341)! This solution is more inline with Dagger 2.11. It makes use of  `AndroidInjector` and `AndroidInjector.Builder` instead of manually declaring the `inject(App app)` and `@BindsInstance` methods.

```java
@Singleton
@Component(modules = AppModule.class)
interface AppComponent extends AndroidInjector<App> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<App> {
    }
}

@Module(includes = AndroidInjectionModule.class)
abstract class AppModule {

    @Binds
    @Singleton
    // Singleton annotation isn't necessary (in this case since Application instance is unique)
    // but is here for convention.
    abstract Application application(App app);
   ...
}

public class App extends Application implements HasActivityInjector {

    @Override
    public void onCreate() {
        super.onCreate();
        DaggerAppComponent.builder().create(this).inject(this);
    }
    ...
}
```

Thanks for this @chronvas! I'd definitely like to have this part merged in.